tools/osbuild-depsolve-dnf5

```diff
@@ -0,0 +1,500 @@
+#!/usr/bin/python3
+# pylint: disable=invalid-name
+
+"""
+A JSON-based interface for depsolving using DNF.
+
+Reads a request through stdin and prints the result to stdout.
+In case of error, a structured error is printed to stdout as well.
+"""
+import json
+import os
+import sys
+import tempfile
+from datetime import datetime
+
+import libdnf5 as dnf5
+from libdnf5.base import GoalProblem_NO_PROBLEM as NO_PROBLEM
+from libdnf5.common import QueryCmp_CONTAINS as CONTAINS
+from libdnf5.common import QueryCmp_EQ as EQ
+from libdnf5.common import QueryCmp_GLOB as GLOB
+
+
+# XXX - Temporarily lifted from dnf.rpm module
+def _invert(dct):
+    return {v: k for k in dct for v in dct[k]}
+
+
+_BASEARCH_MAP = _invert({
```

Comment:
Pardon my ignorance here, where does this map come from? Maybe a comment with a reference where it's from or how it is generated. That probably also answers my next question if we need all of it, i.e. alpha is quite dead for some time already as is sparc AIUI.

Reply:
Looking at this again I feel this is really not something that libdnf5 should burden the use of the library with, i.e. it should either be default or an optional `default_archmap={True,False}` or something. Maybe worth an upstream bugreport? Take it with a grain of salt though, there is a lot of context in the dnf world that I do not have :)

Reply:
This is used for mapping arch strings to the `$basearch` string you see in rpm repo urls. This is used by the rpm repositories, either as part of a path, or by the mirrorlink/metalink servers. When they wrote dnf5 they didn't expose this map in the new code so I lifted it from the old `dnf.rpm` code. It's pretty static and if it ever becomes available upstream again I'll switch it back to using a better source.